### ISSUE TITLE
Extensions, a spec.

### ISSUE THREAD

**Issue by AceLikesGhosts** [AceLikesGhosts/kyusei#9]
# Extensions Spec

An extension, or module, would provide extra functionality to the Kyusei client instance; 
It will provide a way to register commands, and events, as an external module.

### Examples:

Defining a new extension:
```ts
// src/extension.ts
import {  makeExtension } from 'kyusei';
import { join } from 'path'; 

export default makeExtension({
	name: 'Extension name',
	description: 'What the extension does',
	// an array of sub extensions.
	subextensions: [],
	commands: {
		path: join(__dirnname, 'commands')
	}
});

// src/commands/ExampleCommand.ts
import { CacheType, ChatInputCommandInteraction } from 'discord.js';
import { Command, CommandBase } from '../../../../src';

// The command's metadata, what we need to register the command.
@Command({ name: 'example', description: 'Fill in example command' })
export default class ExampleCommandimplements CommandBase
{
    // What we should do when the command is ran.
    async execute(interaction: ChatInputCommandInteraction<CacheType>)
    {
        interaction.reply('ran');
    }
}
```

How to use from a Kyu instance: 
```ts
// src/index.ts
import Kyu from 'kyusei';
import extensionName from './extension';

const client: Kyu<true> = new Kyu({
	// discord.js data & kyu options,
	extensions: [
		extensionName
	]
});

client.init('token');
```

**Comment by AceLikesGhosts** [comment]
RM: Subextension idea, would be stupid.

**Comment by AceLikesGhosts** [comment]
IMPLed and merged in #11
